File: udemy/stack/stack.py
from typing import Any


class Stack(object):
    def __init__(self) -> None:
        self.stack = []

    def push(self, data) -> None:
        self.stack.append(data)

    def pop(self) -> Any:
        if self.stack:
            return self.stack.pop()


if __name__ == '__main__':
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.pop()
    stack.push(3)
    print(stack.stack)

File: udemy/stack/stack.ts
class Stack {
  stack: number[];

  constructor() {
    this.stack = [];
  }

  push(val: number) {
    this.stack.push(val);
  }

  pop() {
    if (this.stack) {
      return this.stack.pop();
    }
  }
}

const stack = new Stack();
stack.push(1);
stack.push(2);
stack.pop();
stack.push(4);
stack.push(3);
console.log(stack.stack);
